raudelb/session05/mainroom_updated.py

```diff
@@ -0,0 +1,89 @@
+__author__ = 'raudel'
+
+
+def send_thanks(my_list):
+    donors_name = []
+
+    for i in my_list:
+        donors_name.append(i[0])
+
+    try:
+
+        full_name = input('Please, enter the Full Name:')
+
+        if full_name == 'list':
+            print('\n----------------------------')
+            print(' Donor Name')
+            print('----------------------------')
+            for i in my_list:
+                print(i[0])
+            print('-------------\n')
+
+        elif full_name not in donors_name:
+
+            t = (full_name, [])
+            my_list.append(t)
+
+        elif full_name in donors_name:
+
+            try:
+
+                donation_amount = int(input('Please, enter the donation amount:'))
+
+                for i in my_list:
+                    if i[0] == full_name:
+                        temp = i[1].append(donation_amount)
+                        print('Email: Thank you %s for your donation of %s $ ' % (full_name, donation_amount))
+
+            except (EOFError, ValueError) as error_msg:
+
+                print('Sorry, you entered a wrong type that produced an ', error_msg )
+
+    except EOFError as error_msg:
```

Comment:
If you are trying to catch any general exception which I think is what you are trying to do here based on the print statement you can do `except Exception as error_msg`.  Also, you probably want to do something other than just print out a message.  Perhaps redirect to the main program or just `return`